### PR TITLE
[KMSv2] apiserver/kmsv2: mv Service interface into kmsv2

### PR DESCRIPTION
**Pull request by ibihim**
Signed-off-by: Krzysztof Ostrowski <kostrows@redhat.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Removes duplicate kms v2 service interface.

It got copied into k8s.io/kms as a first step and it is now being remove from k8s.io/apiserver (but imported).

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP] https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3299-kms-v2-improvements
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope_test.go [hunk 1]
```diff
@@ -30,6 +30,8 @@ import (
 	"k8s.io/apiserver/pkg/storage/value"
 	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
 	kmstypes "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/v2alpha1"
+
```
Comment (aramase) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope_test.go:33]:
```suggestion
```

**Review by aramase — COMMENTED** [review]
Just one small nit!

`pull-kubernetes-unit` failure might just be a flake.

**Review by enj — CHANGES_REQUESTED** [review]

**Pull request merged**